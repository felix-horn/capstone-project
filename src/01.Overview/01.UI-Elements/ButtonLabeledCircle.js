import PropTypes from 'prop-types'
import styled from 'styled-components/macro'
import ButtonCircle from './ButtonCircle'

ButtonLabeledCircle.propTypes = {
  onClick: PropTypes.func.isRequired,
  title: PropTypes.string.isRequired,
  children: PropTypes.element.isRequired,
  variant: PropTypes.string,
  className: PropTypes.string,
}

export default function ButtonLabeledCircle({
  onClick,
  title,
  children,
  variant,
  className,
}) {
  return (
    <ButtonLayout
      onClick={onClick}
      className={className}
      data-testid="labeled-circle-button"
    >
      {title}
      <ButtonCircle /* className="small" */ variant={variant}>
        {children}
      </ButtonCircle>
    </ButtonLayout>
  )
}

const ButtonLayout = styled.div`
  display: flex;
  justify-content: end;
  align-items: center;
  gap: 20px;
  color: var(--almost-black);
  font-weight: 400;
  font-size: 0.8rem;
  text-decoration: none;

  &.primary {
    gap: 15px;
  }
`
