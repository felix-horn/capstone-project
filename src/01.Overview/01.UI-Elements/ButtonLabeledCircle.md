```js
import MenuIcon from '@material-ui/icons/Menu'
;<ButtonLabeledCircle title="Button Lable" onClick={() => {}}>
  <MenuIcon />
</ButtonLabeledCircle>
```
