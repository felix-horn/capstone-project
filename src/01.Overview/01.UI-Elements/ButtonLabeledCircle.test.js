import { render } from '@testing-library/react'
import user from '@testing-library/user-event'
import ButtonLabeledCircle from './ButtonLabeledCircle'
import AddShopIcon from '@material-ui/icons/PostAdd'

describe('ButtonLabeledCircle', () => {
  it('renders correctly', () => {
    const onClickMock = jest.fn()
    const { container } = render(
      <ButtonLabeledCircle onClick={onClickMock} title="button title">
        <AddShopIcon />
      </ButtonLabeledCircle>
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  it('calls function onClick', () => {
    const onClickMock = jest.fn()
    const { getByTestId } = render(
      <ButtonLabeledCircle onClick={onClickMock} title="button title">
        <AddShopIcon />
      </ButtonLabeledCircle>
    )
    const button = getByTestId('labeled-circle-button')
    user.click(button)
    expect(onClickMock).toHaveBeenCalledTimes(1)
  })

  it('shows the correct button label', () => {
    const onClickMock = jest.fn()
    const { getByTestId } = render(
      <ButtonLabeledCircle onClick={onClickMock} title="button title">
        <AddShopIcon />
      </ButtonLabeledCircle>
    )
    const button = getByTestId('labeled-circle-button')
    expect(button.firstChild.textContent).toEqual('button title')
  })

  it('displays the correct height for variant "small"', () => {
    const onClickMock = jest.fn()
    const { getByTestId } = render(
      <ButtonLabeledCircle
        variant="small"
        onClick={onClickMock}
        title="button title"
      >
        <AddShopIcon />
      </ButtonLabeledCircle>
    )
    const button = getByTestId('button')
    const style = window.getComputedStyle(button)
    expect(style.height).toBe('40px')
  })

  xit('displays the correct color for variant "primary"', () => {
    const onClickMock = jest.fn()
    const { getByTestId } = render(
      <ButtonLabeledCircle
        variant="primary"
        onClick={onClickMock}
        title="button title"
      >
        <AddShopIcon />
      </ButtonLabeledCircle>
    )
    const button = getByTestId('button')
    // console.log(button.getComputedStyle)
    const style = window.getComputedStyle(button)
    //    backgroundColor is not available
    // console.log(style.backgroundColor)
    expect(style.backgroundColor).toBe('var(--CTA-blu)')
    // expect(button).toHaveStyle(`background: var(--CTA-blu)`)
  })
})
