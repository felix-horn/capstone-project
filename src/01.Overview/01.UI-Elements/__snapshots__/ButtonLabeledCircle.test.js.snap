// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonLabeledCircle renders correctly 1`] = `
.c1 {
  box-shadow: var(--strong-box-shadow);
  border-radius: 100%;
  height: 50px;
  width: 50px;
  background-color: var(--white);
  display: grid;
  place-items: center;
  color: var(--CTA-blue);
}

.c1.primary {
  background-color: var(--CTA-blue);
  color: var(--white);
}

.c1.small {
  height: 40px;
  width: 40px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: end;
  -ms-flex-pack: end;
  justify-content: end;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 20px;
  color: var(--almost-black);
  font-weight: 400;
  font-size: 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0.primary {
  gap: 15px;
}

<div
  class="c0"
  data-testid="labeled-circle-button"
>
  button title
  <div
    class="c1"
    data-testid="button"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M17 19.22H5V7h7V5H5c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2v-7h-2v7.22z"
      />
      <path
        d="M19 2h-2v3h-3c.01.01 0 2 0 2h3v2.99c.01.01 2 0 2 0V7h3V5h-3V2zM7 9h8v2H7zM7 12v2h8v-2h-3zM7 15h8v2H7z"
      />
    </svg>
  </div>
</div>
`;
